libp2p/protocols/secure/secio.nim

```diff
@@ -461,7 +461,7 @@ method init(s: Secio) {.gcsafe.} =
   s.codec = SecioCodec
   s.handler = handle
 
-method secure*(s: Secio, conn: Connection): Future[Connection] {.async, gcsafe.} =
+method secure*(s: Secio, conn: Connection, outgoing: bool): Future[Connection] {.async, gcsafe.} =
```

Comment:
@sinkingsugar I still don't think we should modify `secure` to take the initiator parameter, the point is that when calling `secure` you are implicitly the initiator and when `handler` is triggered you are implicitly the receiver. I still don't understand the need for this flag.

Reply:
beacon chain does not need it indeed, but the way the testswitch is written needs this as both parties are 2 initiators.. That said, I will check why secio doesn't hang and worst case add it to `newNoise` and call it a day since it's something that indeed won't be used much cept for that test or so.

Reply:
So indeed it seems that secio does not need this as internally during handshake there is no real difference between initiator or not, in noise it's 2 different paths, that said I will add it on the noise struct initiator so it's easier to opt out

Reply:
Btw are we really sure we want `type[T].init` pattern on `ref` types?
It kinda makes it hard to track "references" while writing imho if the name is shared with value types. ping @arnetheduck